### DISCUSSION TITLE
atomWithHash problem serializing "all" key

### DISCUSSION THREAD

**Discussion by hobbio** [pmndrs/jotai#1329] [Q&A]
Using atomWithHash for saving state in url the key "all" is serialized as undefined.

I have the next config on a react project 

 
`

const statusAtom = useWithHash("status", "open")

const Component  = () => {

   const [status, setStatus] = useAtom(statusAtom)

  const handleKeyChange = (key) => {

    setStatus(key)
}
  
  return <Tab onChange={handleKeyChange}>
    <TabPane key="open"></TabPane> 
    <TabPane key="closed"></TabPane>
    <TabPane key="all"></TabPane>
<Tab>

}
`

Everything  works great until "all" tab is selected  for  a strange reason when I click on "all" tab  the URL encoding (serialization) return "undefined"  but the state works fine "all" it just happend with the "all" word,  if we change the key to "all-x" the url is updated correctly

When status is all 

https://exampleurs/page#status=undefined 

When status is all-x 

https://exampleurs/page#status="all-x"

I think there is a internal conflict with the all word in jotai library

**Comment by dai-shi** [comment]
By default, it uses `JSON.parse` to deserialize. You can provide your own deserialize function for your need.

It would be nice if docs cover this, including type checking.

**Reply by hobbio** [reply]
I understand we can provide a custom deserialize function. 

Im trying to understand why  "all" value did not match with the JSON.parse default deserialize function. But still the probles is not deserializing the content, it is on serializing the content, because the status is still "all" and its working normally but the URL is printed incorrectly.  Maybe is a bug in the library

**Reply by dai-shi** [reply]
Oh, I misunderstood your point. Sorry about that. So, it's an issue with the _value_ "all".
Could you create a minimal codesandbox reproduction?
